docker-entrypoint.sh

```diff
@@ -141,47 +190,91 @@ trustProxyCertificate() {
     fi
 }
 
-trustRootCertificate() {
-    $certificateManagerDir/certificate-manager.sh trust-java-cert \
-                        --store $truststoreFile \
-                        --password changeit \
-                        --cert $securityDir/root.crt \
-                        --certAlias hub-root
+createKeystore() {
+    certKey=$securityDir/$serverCertName.key
+    certFile=$securityDir/$serverCertName.crt
+    if [ -f $dockerSecretDir/WEBSERVER_CUSTOM_CERT_FILE ] && [ -f $dockerSecretDir/WEBSERVER_CUSTOM_KEY_FILE ];
+    then
+        certKey="${dockerSecretDir}/WEBSERVER_CUSTOM_KEY_FILE"
+        certFile="${dockerSecretDir}/WEBSERVER_CUSTOM_CERT_FILE"
 
+        echo "Custom webserver cert and key found"
+    	echo "Using $certFile and $certKey for webserver"
+    fi
+    # Create the keystore with given private key and certificate.
+    echo "Attempting to create keystore."
+    $certificateManagerDir/certificate-manager.sh keystore \
+                                             --outputDirectory $securityDir \
+                                             --outputFile $keyStoreFile \
+                                             --password changeit \
+                                             --keyAlias $APPLICATION_NAME \
+                                             --key $certKey \
+                                             --cert $certFile
     exitCode=$?
     if [ $exitCode -eq 0 ];
     then
-        echo "Successfully imported Hub root certificate into Java truststore."
+        chmod 644 $keystoreFilePath
     else
-        echo "Unable to import Hub root certificate into Java truststore (Code: $exitCode)."
+        echo "Unable to create keystore (Code: $exitCode)."
         exit $exitCode
     fi
 }
 
+importBlackDuckSystemCertificateIntoKeystore() {
+  $certificateManagerDir/certificate-manager.sh trust-java-cert \
+                        --store $keystoreFilePath \
+                        --password changeit \
+                        --cert $securityDir/blackduck_system.crt \
+                        --certAlias blackduck_system
+
+    exitCode=$?
+    if [ $exitCode -eq 0 ];
+    then
+        echo "Successfully imported BlackDuck system certificate into Java truststore."
```

Comment:
This log is confusing, since we are importing the certificate into the Keystore. Is the log wrong? or should we import the certificate into the trust store?

Reply:
yes copy paste error here.  I will update it.